src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -219,163 +224,108 @@ public bool HasMultipleConstructorsOfSameArity(ref GeneratorExecutionContext con
+                var d = Diagnostic.Create(DiagnosticRules.ArraySignature_SystemArrayRule, loc, typeId, propId);
+                found |= SignatureContainsTypeName(ref context, qualifiedTypes, "System.Array", d);
                 
-                found |= ArrayIsntOneDim(prop.DescendantNodes().OfType<ArrayTypeSyntax>(), ref context, classId, propId, loc);
+                var types = prop.DescendantNodes().OfType<IdentifierNameSyntax>(); 
+                var d2 = Diagnostic.Create(DiagnosticRules.ArraySignature_SystemArrayRule, loc, typeId, propId);
+                found |= SignatureContainsTypeName(ref context, types, "Array", d2);
+                
+                found |= ArrayIsntOneDim(prop.DescendantNodes().OfType<ArrayTypeSyntax>(), ref context, typeId, propId, loc);
             }
             return found;
         }
 
         /// <summary>
-        /// Look at all the array types and if any are of the form [][]+ or [,+] then raise the corresponding diagnostic and return true
-        /// </summary>
-        /// <param name="arrTypes"></param>
-        /// <param name="context"></param>
-        /// <param name="typeIdentifier">The type the array lives in</param>
-        /// <param name="fieldId">The code the array is a part of the signature for; e.g. property or method</param>
-        /// <param name="loc"></param>
+        /// Look at all the array types and if any are of the form [][]+ or [,+] then raise the corresponding diagnostic and return true /// </summary>
+        /// <param name="arrTypes"></param><param name="context"></param><param name="typeIdentifier">The type the array lives in</param>
+        /// <param name="fieldId">The code the array is a part of the signature for; e.g. property or method</param><param name="loc"></param>
         /// <returns>True iff any of the array types given are multidimensional or jagged</returns>
         private bool ArrayIsntOneDim(IEnumerable<ArrayTypeSyntax> arrTypes, ref GeneratorExecutionContext context, SyntaxToken typeIdentifier, SyntaxToken fieldId, Location loc)
         {
             foreach (var arrType in arrTypes)
             {
                 var brackets = arrType.DescendantNodes().OfType<ArrayRankSpecifierSyntax>();
-                if (brackets.Count() > 1) // [][]+ ?
+                // [][]+ ?
+                if (brackets.Count() > 1) 
                 {
-                    context.ReportDiagnostic(Diagnostic.Create(DiagnosticRules.ArraySignature_JaggedArrayRule, loc, fieldId, typeIdentifier));
-                    return true; // could do or-eq on a `found` boolean instead of exiting as soon as we see invalid...
+                    Report(ref context, DiagnosticRules.ArraySignature_JaggedArrayRule, loc, fieldId, typeIdentifier);
+                    return true;
                 }
-                // [,_] ? 
+                // [,+] ? 
                 else if (brackets.Count() == 1 && brackets.First().ToString().Contains(","))

```

Comment:
what about white space?

Reply:
Made a test with an array like `[ , , ]` and we raise diagnostic for it, so think this is fine...